internal/security-governance-system/webapp/src/main/java/org/wso2/security/tools/sgs/webapp/service/ScanService.java

```diff
@@ -0,0 +1,233 @@
+        } catch (URISyntaxException e) {
+            logger.error("Unable to build the start scan URI", e);
+        } catch (IOException e) {
+            logger.error("Unable to start the scan", e);
+        } finally {
+            try {
+                if (client != null) {
+                    client.close();
+                }
+            } catch (IOException e) {
+                logger.error("Unable to close the http connection with scan manager", e);
+            }
+        }
+        return responseStatus;
+    }
+
+    private static String convertStringArrayToString(String[] strArr, String delimiter) {
+        StringBuilder sb = new StringBuilder();
+        for (String str : strArr)
+            sb.append(str).append(delimiter);
+        return sb.substring(0, sb.length() - 1);
+    }
+
+    /**
+     * Get the list of scans in the pool
+     *
+     * @return
+     */
+    public ArrayList<Scan> getScans() {
+
+        MultiValueMap headerMap = new LinkedMultiValueMap();
+        Map<String, Object> paramMap = new HashMap<>();
+        ArrayList<Scan> scanList = new ArrayList<>();
+        HttpRequestHandler httpRequestHandler = new HttpRequestHandler();
+
+        try {
+            URI uri = (new URIBuilder())
+                    .setHost(sgsWebappConfiguration.getScanManagerHost())
+                    .setPort(sgsWebappConfiguration.getScanManagerPort())
+                    .setScheme(SCHEME).setPath(sgsWebappConfiguration.getScansURI())
+                    .build();
+
+            ResponseEntity<String> responseEntity = httpRequestHandler.sendGET(uri.toString(), headerMap, paramMap);
+            JSONArray responseArray = new JSONArray(responseEntity.getBody());
+
+            for (int arrayIndex = 0; arrayIndex < responseArray.length(); arrayIndex++) {
+                JSONObject jsonObject = responseArray.getJSONObject(arrayIndex);
+                Scan scan = new Scan();
+                scan.setId(jsonObject.get(ID).toString());
+                scan.setName(jsonObject.get(NAME).toString());
+                scan.setOwner(jsonObject.get(OWNER).toString());
+                scan.setStatus(jsonObject.get(STATUS).toString());
+                scanList.add(scan);
+            }
+
+        } catch (RestClientException | JSONException | URISyntaxException e) {
+            logger.error("Unable to get the scans", e);
+        }
+        return scanList;
+    }
+
+    /**
+     * Stop a scan
+     *
+     * @param id
+     * @return
+     */
+    public ResponseEntity stopScan(String id) {
+
+        ResponseEntity<String> responseEntity = null;
+        MultiValueMap headerMap = new LinkedMultiValueMap();
+        headerMap.add("Content-Type", "application/json");
+        Map<String, Object> paramMap = new HashMap<>();
+        HttpRequestHandler httpRequestHandler = new HttpRequestHandler();
+
+        paramMap.put(ID, id);
+
+        try {
+            URI uri = (new URIBuilder())
```

Comment:
This URL building can be done inside the configuration class itself. That class could contain a getter that returns the required URI.